pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -385,6 +385,10 @@ stdenv.mkDerivation ({
     "--enable-dwarf-unwind"
     "--with-libdw-includes=${lib.getDev elfutils}/include"
     "--with-libdw-libraries=${lib.getLib elfutils}/lib"
+  ] ++ lib.optionals targetPlatform.isDarwin [
+    # Darwin uses llvm-ar. GHC will try to use `-L` with `ar` when it is `llvm-ar`
+    # but it doesn’t currently work. See https://gitlab.haskell.org/ghc/ghc/-/issues/23188.
```

Comment:
Can you also link https://github.com/haskell/cabal/issues/8882?

Reply:
Sure. I’ll add the Cabal issue as well when I’m back at our room and can push an update.

Reply:
Done. I added the Cabal issue as well.